azurelinuxagent/common/utils/networkutil.py

```diff
@@ -200,21 +248,27 @@ def __execute_cmd(cmd):
 
     @staticmethod
     def add_iptables_rules(wait, dst_ip, uid):
-
+        # The order in which the below rules are added matters for the ip table rules to work as expected
         AddFirewallRules.__raise_if_empty(dst_ip, "Destination IP")
         AddFirewallRules.__raise_if_empty(uid, "User ID")
 
-        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.__ACCEPT_COMMAND, dst_ip, uid)
+        accept_rule_dns_tcp_rule = AddFirewallRules.get_iptables_accept_dns_tcp_request_command(wait, AddFirewallRules.get_insert_command(),dst_ip)
```

Comment:
NIT: 
> PEP 8: E231 missing whitespace after ','

Reply:
If you're adding this rule at the beginning of the chain, it doesn't really need Insert right? Append would work too no?

Reply:
Agreed, When it comes to iptable rules it doesn't matter. But for firewalld since its persistent we have to use insert. Earlier I used insert to be consistent with both iptable and firewalld . Now changed it to append for iptable